### ISSUE TITLE
Expose deploymentId of created or updated application to be used in deployment monitoring

### ISSUE THREAD

**Issue by krystiannowak** [willroden/marathon-client#3]
There are certain situations when after requesting a deployment of an application to the Marathon cluster we need to wait certain amount of time till the deployment finished before we can query another API call in the context of this application e.g. some metadata (hostname, port) about the deployment. We were trying to use this scenario - see: https://github.com/holidaycheck/marathon-maven-plugin/pull/1#issuecomment-121939044 - but since we do not have currently support to obtain _deploymentId_ when creating or updating an application, we ended up with heuristics like this one: https://github.com/holidaycheck/marathon-maven-plugin/blob/fac26f9d6f837604f911ac1513ac4d12479bce5c/src/main/java/com/holidaycheck/marathon/maven/AppTasksMojo.java#L78 (contributed by @bsneade)

In order to make it more reliable and not to use timeout-based approach some changes would need to be applied (still their validity is to be discussed probably) to this project to support the approach based on _deploymentId_ monitoring.

Application creation API:
https://mesosphere.github.io/marathon/docs/rest-api.html#post-/v2/apps
returns:

```
"deployments": [
    {
        "id": "f44fd4fc-4330-4600-a68b-99c7bd33014a"
    }
]
```

element

Application update API:
https://mesosphere.github.io/marathon/docs/rest-api.html#put-/v2/apps/{appid}
returns:

```
"deploymentId": "83b215a6-4e26-4e44-9333-5c385eda6438"
```

element.

This can be used by checking the result of this call:
https://mesosphere.github.io/marathon/docs/rest-api.html#get-/v2/deployments
and looking for _deploymentId_ in:

```
"affectedApps": [
    "/test"
],
"id": "867ed450-f6a8-4d33-9b0e-e11c5513990b",
```

element.

Currently in marathon-client we have https://github.com/willroden/marathon-client/blob/5c33bfc8296678aef95ef9a19c79498a64c2a306/src/main/java/mesosphere/marathon/client/Marathon.java#L36
for application creation and https://github.com/willroden/marathon-client/blob/5c33bfc8296678aef95ef9a19c79498a64c2a306/src/main/java/mesosphere/marathon/client/Marathon.java#L39 for updating the application.

We are missing _deploymentIds_ in returned types, e.g. creation returns https://github.com/willroden/marathon-client/blob/5c33bfc8296678aef95ef9a19c79498a64c2a306/src/main/java/mesosphere/marathon/client/model/v2/App.java
(doesn't contain _deploymentIds_) and update return type is void currently.

We already have deployments request https://github.com/willroden/marathon-client/blob/5c33bfc8296678aef95ef9a19c79498a64c2a306/src/main/java/mesosphere/marathon/client/Marathon.java#L70
and the model returned https://github.com/willroden/marathon-client/blob/5c33bfc8296678aef95ef9a19c79498a64c2a306/src/main/java/mesosphere/marathon/client/model/v2/Deployment.java#L10 contains ID of deployment to match on.

**Comment by WillAbides** [comment]
I've been away and just saw this.  I'll look at it in detail shortly.

**Comment by WillAbides** [comment]
Thanks to @bsnead's pull request.  This is implemented in 0.4.8 which should be in maven central very shortly.
